internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -323,26 +289,26 @@ func WithPollInterval(after time.Duration) ReconcilerOption {
 func NewReconciler(m manager.Manager, of resource.CompositeClaimKind, with resource.CompositeKind, o ...ReconcilerOption) *Reconciler {
 	c := unstructured.NewClient(m.GetClient())
 	r := &Reconciler{
-		client: resource.ClientApplicator{
-			Client:     c,
-			Applicator: resource.NewAPIPatchingApplicator(c),
-		},
-		newClaim: func() resource.CompositeClaim {
+		client: c,
+		newClaim: func() *claim.Unstructured {
 			return claim.New(claim.WithGroupVersionKind(schema.GroupVersionKind(of)))
 		},
-		newComposite: func() resource.Composite {
+		newComposite: func() *composite.Unstructured {
 			return composite.New(composite.WithGroupVersionKind(schema.GroupVersionKind(with)))
 		},
-		composite: defaultCRComposite(c),
-		claim:     defaultCRClaim(c),
-		log:       logging.NewNopLogger(),
-		record:    event.NewNopRecorder(),
+		composite:  defaultCRComposite(c),
+		claim:      defaultCRClaim(c),
+		log:        logging.NewNopLogger(),
+		record:     event.NewNopRecorder(),
+		fieldOwner: client.FieldOwner(fmt.Sprintf("crossplane-claim-controller/%s", strings.ToLower(schema.GroupVersionKind(of).GroupKind().String()))),
```

Comment:
> I don't see a value with including group/kind into the field owner. 

Having the same owner name shared by all claim controllers could lead to implicit shared ownership between controllers and degrades the user experience. We have a controller per specific claim type, and browsing through the kubernetes code base shows that controllers set an unique field manager name, a few examples:

* https://github.com/kubernetes/kubernetes/blob/d61cbac69aae97db1839bd2e0e86d68f26b353a7/pkg/controller/validatingadmissionpolicystatus/controller.go#L38
* https://github.com/kubernetes/kubernetes/blob/d61cbac69aae97db1839bd2e0e86d68f26b353a7/pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go#L141
* https://github.com/kubernetes/kubernetes/blob/d61cbac69aae97db1839bd2e0e86d68f26b353a7/pkg/controlplane/controller/defaultservicecidr/default_servicecidr_controller.go#L45
* https://github.com/kubernetes/kubernetes/blob/d61cbac69aae97db1839bd2e0e86d68f26b353a7/pkg/controller/resourceclaim/controller.go#L66

Including group/kind as a part of the field owner name is an easy way to establish the name uniqueness and at the same time it gives an user a clear information about the field ownership, e.g. when user performs something like:

```sh
kubectl get foo --show-managed-fields
```

We use similar strategies when declaring controller name for the log context. Ofcourse, we can polish the proposed convention if needed, but we should keep an unique name per claim controller.

> Proposing just to use apiextensions.crossplane.io/claim also to be consistent with the Composite's: 

Based on the above explanation, I was about to propose the convention in followup PRs that are about fixing composite -> managed resources synchronization, i.e. we should introduce an unique name per composite controller.

Reply:
> Having the same owner name shared by all claim controllers could lead to implicit shared ownership between controllers and degrades the user experience

I don't think this is practically possible. Each controller [works](https://github.com/crossplane/crossplane/blob/6756027ce951286d0c07ea38b5c1db141e89136f/internal/controller/apiextensions/claim/reconciler.go#L331) with its own type and there is no way for one claim controller to operate on another gvk. So, my preference would be to keep it simple.

> an user a clear information about the field ownership, e.g. when user performs something like:

I think it would still be clear enough if/when we just say this is owned by claim controller. No one would be confused about which claim's controller is operating on that claim, obviously, the claim controller for that GVK.

Reply:
> I think it would still be clear enough if/when we just say this is owned by claim controller. No one would be confused about which claim's controller is operating on that claim, obviously, the claim controller for that GVK.

Claim controller creates composites as well and owns a good part of it. As we discussed offline, it might be useful for users when debugging to understand which claim controller owns certain fields.

Reply:
done, reverted to the approach you are proposing.